### PR TITLE
Bkz2 get pruning

### PR DESCRIPTION
**Pull request by lducas**
Solve issue #84

### CODE REVIEW

File: src/fpylll/algorithms/bkz2.py [hunk 1]
```diff
@@ -17,13 +17,18 @@ def __init__(self, A):
...
+        strategy = params.strategies[block_size]
+        radius = self.M.get_r(kappa, kappa) * self.lll_obj.delta
 
-        radius, re = self.M.get_r_exp(kappa, kappa)
-        root_det = self.M.get_root_det(kappa, kappa + block_size)
-        gh_radius, ge = adjust_radius_to_gh_bound(radius, re, block_size, root_det, 1.0)
-        return strategy.get_pruning(radius  * 2**re, gh_radius * 2**ge)
+        # Don't bother computing expensive gh for small blacksizes
+        if not (params.flags & BKZ.GH_BND and block_size > 30):
+            return radius, strategy.get_pruning(radius, 100 * radius)
```
Comment (malb) [inline src/fpylll/algorithms/bkz2.py]: I don't understand the `100*radius` logic

Comment (lducas) [inline src/fpylll/algorithms/bkz2.py]: Admittedly, I could have written 4 (the square of 2). The goal is merely to take the pruning bound computed for the largest possible gh_factor.

Comment (malb) [inline src/fpylll/algorithms/bkz2.py]: Sorry, still being daft: You are picking the pruning parameters with the largest radius whenever `BKZ.GH_BND` is not set *or* when the block size < 30. For example, running BKZ-80 with `BKZ.GH_BND` set to `False` would mean always select the pruning parameters with the largest radius.

Comment (lducas) [inline src/fpylll/algorithms/bkz2.py]: You are not daft. Indeed, I only meant that for block-size < 30. It must be ammended. Maybe I can check the effect of always doing it (ie computing the gh heuristic) and see if there is no slowdown...

File: src/fpylll/algorithms/bkz2.py [hunk 2]
```diff
@@ -17,13 +17,19 @@ def __init__(self, A):
...
-        strategy = param.strategies[block_size]
+    def get_pruning(self, kappa, block_size, params, tracer=dummy_tracer):
+        strategy = params.strategies[block_size]
+        radius = self.M.get_r(kappa, kappa) * self.lll_obj.delta
 
-        radius, re = self.M.get_r_exp(kappa, kappa)
-        root_det = self.M.get_root_det(kappa, kappa + block_size)
-        gh_radius, ge = adjust_radius_to_gh_bound(radius, re, block_size, root_det, 1.0)
-        return strategy.get_pruning(radius  * 2**re, gh_radius * 2**ge)
+        # Don't bother computing expensive gh for small blacksizes
```
Comment (malb) [inline src/fpylll/algorithms/bkz2.py]: This comment seems outdated now, also note "blacksizes" → "blocksizes"

Comment (lducas) [inline src/fpylll/algorithms/bkz2.py]: Thx for your vigilance...

**Pull request merged**